src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,8 +1,145 @@
+
+        User user2 = new User();
+        user2.setLogin("Parmezan");
+        user2.setPassword("12345678910");
+        user2.setAge(50);
+        result = registrationService.register(user2);
+        assertEquals(result, user2);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        User user = null;
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_addIdenticalUsers_notOk() {
+        User user = new User();
+        user.setLogin("Antonio");
+        user.setPassword("12345678");
+        user.setAge(23);
+
+        registrationService.register(user);
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_addYoungUser_notOk() {
+        User user1 = new User();
+        user1.setLogin("Martin77");
+        user1.setPassword("12345678");
+        user1.setAge(15);
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user1));
+
+        User user2 = new User();
+        user2.setLogin("Kartoplia");
+        user2.setPassword("12345678");
+        user2.setAge(0);
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user2));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("12345678");
+        user.setAge(23);
+
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("Arturitto");
+        user.setPassword(null);
+        user.setAge(72);
+
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword("12345678");
+        user.setAge(30);
+
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("Login");
+        user.setPassword("12345678");
+        user.setAge(45);
+
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("Hello!");
+        user.setPassword("1234");
+        user.setAge(19);
+
+        assertThrows(RegistrationFailureException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_allParametersNull_notOk() {
+        User user = new User();
+        user.setLogin(null);
+        user.setPassword(null);
+        user.setAge(null);
```

Comment:
wouldn't say that we need that case because null login and lull password are already tested separately, we are only missing a separate null check of age test.
What else is interesting here:
* The exception about null age will never be thrown as you are expecting it, and password in fact too, because it will always check the login first, and thus will always throw the same exception about invalid login

We should strive for as deep & detailed tests as possible, to guarantee that the code we wrote works **exactly** as we think it does